dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -432,6 +443,21 @@ normalizingAnExpressionDoesntChangeItsInferredType expression =
     filterOutEmbeds :: Typer a
     filterOutEmbeds _ = Const Sort -- This could be any ill-typed expression.
 
+injectThenInterpretIsIdentity
+    :: forall a. (Inject a, Interpret a, Eq a, Typeable a, Arbitrary a, Show a)
+    => Proxy a
+    -> (String, Property, QuickCheckTests)
+injectThenInterpretIsIdentity p =
+    ( "Injecting then Interpreting is identity for " <> show (typeRep p)
+    , Test.QuickCheck.property (prop :: a -> Property)
+    , QuickCheckTests 1000
+    )
+  where
+    prop a = monadicIO $ do
+        a' <- run . input auto . Text.pack . show . prettyExpr . embed inject $ a
```

Comment:
Why do you pretty-print and re-parse the embedded expressions?

I you'd simply `extract` the `embed`ded `Expr`, we wouldn't need `IO`…

Reply:
Oh man... I feel I'm doing so many obvious mistakes. Thank you!

Reply:
No worries! :)